processing/src/main/java/org/apache/druid/segment/DimensionSelector.java

```diff
@@ -164,6 +166,27 @@ static DimensionSelector constant(@Nullable final String value, @Nullable final
     }
   }
 
+  static DimensionSelector multiConstant(@Nullable final List<String> values)
+  {
+    if (values == null || values.isEmpty()) {
+      return NullDimensionSelectorHolder.NULL_DIMENSION_SELECTOR;
+    } else {
+      return new ConstantMultiValueDimensionSelector(values);
+    }
+  }
+
+  static DimensionSelector multiConstant(@Nullable final List<String> values, @Nullable final ExtractionFn extractionFn)
+  {
+    if (extractionFn == null) {
+      return multiConstant(values);
+    } else {
+      if (values == null) {
+        return constant(extractionFn.apply(null));
```

Comment:
What's the rationale behind doing this instead of `multiConstant(Collections.singletonList(extractionFn.apply(null))`?

Reply:
i originally had it like that, but am unsure if `null` and `[null]` are _actually_ the same thing. From the perspective of grouping, they are identical I guess so it probably doesn't matter. To me the most correct thing would be to _not_ apply the extraction at all and just make it a constant null and not a list with a null, but those are sort of the same thing for string columns currently...

Reply:
to add to this, `constant` is a bit better than `multiConstant` because it reports its cardinality, which I guess I could do for `multiConstant` if i build and sort dictionary and do it that way, but didn't because i was lazy. So, unless I rework `multiConstant` to be a bit cooler then i think is better to use `constant`. Should I make `multiConstant` better? I guess it still doesn't answer if `null` is equivalent to `[null]` in all cases or not.

Reply:
added comments to explain everything and made a few behavior modifications to make sure `multiConstant` has consistent behavior with other multi-valued string selectors